pkg-r/R/QueryChat-class.R

```diff
@@ -0,0 +1,661 @@
+          }
+
+          sql_code <- paste(c("```sql", sql, "```"), collapse = "\n")
+
+          shinychat::output_markdown_stream(
+            "sql_code",
+            content = sql_code,
+            auto_scroll = FALSE,
+            width = "100%"
+          )
+        })
+
+        shiny::observeEvent(input$close_btn, label = "on_close_btn", {
+          shiny::stopApp()
+        })
+      }
+
+      app <- shiny::shinyApp(ui, server, enableBookmarking = bookmark_store)
+      tryCatch(shiny::runGadget(app), interrupt = function(cnd) NULL)
+      invisible(chat)
+    },
+
+    #' @description
+    #' Create a sidebar containing the querychat UI.
+    #'
+    #' This method generates a [bslib::sidebar()] component containing the chat
+    #' interface, suitable for use with [bslib::page_sidebar()] or similar layouts.
+    #'
+    #' @param width Width of the sidebar in pixels. Default is 400.
+    #' @param height Height of the sidebar. Default is "100%".
+    #' @param ... Additional arguments passed to [bslib::sidebar()].
+    #'
+    #' @return A [bslib::sidebar()] UI component.
+    #'
+    #' @examples
+    #' \dontrun{
+    #' qc <- QueryChat$new(mtcars, "mtcars")
+    #'
+    #' ui <- page_sidebar(
+    #'   qc$sidebar(),
+    #'   # Main content here
+    #' )
+    #' }
+    sidebar = function(width = 400, height = "100%", ...) {
+      bslib::sidebar(
+        width = width,
+        height = height,
+        class = "querychat-sidebar",
+        ...,
+        self$ui()
+      )
+    },
+
+    #' @description
+    #' Create the UI for the querychat chat interface.
+    #'
+    #' This method generates the chat UI component. Typically you'll use
+    #' `$sidebar()` instead, which wraps this in a sidebar layout.
+    #'
+    #' @param ... Additional arguments passed to [shinychat::chat_ui()].
+    #'
+    #' @return A UI component containing the chat interface.
+    #'
+    #' @examples
+    #' \dontrun{
+    #' qc <- QueryChat$new(mtcars, "mtcars")
+    #'
+    #' ui <- fluidPage(
+    #'   qc$ui()
+    #' )
+    #' }
+    ui = function(...) {
+      mod_ui(self$id, ...)
+    },
+
+    #' @description
+    #' Initialize the querychat server logic.
+    #'
+    #' This method must be called within a Shiny server function. It sets up
+    #' the reactive logic for the chat interface and populates the internal
+    #' server values that are accessed via `$df()`, `$sql()`, and `$title()`.
+    #'
+    #' @param session The Shiny session object.
+    #'
+    #' @return Invisibly returns `NULL`. Access reactive values via `$df()`,
+    #'   `$sql()`, and `$title()` methods.
+    #'
+    #' @examples
+    #' \dontrun{
+    #' qc <- QueryChat$new(mtcars, "mtcars")
+    #'
+    #' server <- function(input, output, session) {
+    #'   qc$server()
+    #'
+    #'   output$data <- renderDataTable(qc$df())
+    #'   output$query <- renderText(qc$sql())
+    #' }
+    #' }
+    server = function(session = shiny::getDefaultReactiveDomain()) {
```

Comment:
Maybe `auto_cleanup` would go here nd in `mod_server()`?

Reply:
`shiny::onStop()` is equivalent to `shiny::onSessionEnded()` when used inside the server function....